hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateMap.java

```diff
@@ -97,19 +97,24 @@ void addPipeline(Pipeline pipeline) throws IOException {
    *
    * @param pipelineID - PipelineID of the pipeline to which container is added
    * @param containerID - ContainerID of the container to add
+   * @param  checkPipelineClosed - check if pipeline is closed
```

Comment:
Maybe this is just confusing for me, I'll let you decide to change this or not.
```suggestion
   * @param  checkPipelineClosed - Whether to check if pipeline is closed before adding the container. If true, we will throw an exception in case of a closed pipeline.
```

Reply:
I think `rejectClosedPipeline` would be a better name to describe the purpose of this parameter.  `check...` is ambiguous, it may as well indicate that the method only accepts closed pipelines...

(Or flip the meaning of the flag and use `allowClosedPipeline`.)